src/test/scala/com/box/castle/core/leader/TaskManagerTest.scala

```diff
@@ -127,6 +134,67 @@ class TaskManagerTest extends Specification with Mockito with Logging with MockT
+      createCommitterConfig(InitialOffset.latest, "falseCommitter", topicsRegexRaw=None, topicsSet=Some(Set("performance"))))
+
+    val taskManager = getTasksManager(committerConfigs)
+
+    "A topic that matches the filter should return true" in {
+      taskManager.matchCommitterTopicFilter(performanceTopic, "kafkaCommitter") shouldEqual true
+      taskManager.matchCommitterTopicFilter(performanceTopic, "esCommitter") shouldEqual true
+      taskManager.matchCommitterTopicFilter(performanceTopic, "falseCommitter") shouldEqual false
+
+      taskManager.matchCommitterTopicFilter(loginTopic, "kafkaCommitter") shouldEqual true
+      taskManager.matchCommitterTopicFilter(loginTopic, "esCommitter") shouldEqual true
+      taskManager.matchCommitterTopicFilter(loginTopic, "falseCommitter") shouldEqual false
+
+    }
+  }
+
+  "matchTopicsRegex" should {
+    val performanceTopic = KafkaTopic("performance", Set(0, 1))
+    val loginTopic = KafkaTopic("login", Set(8, 9))
+    val kafkaTopics = Set(performanceTopic, loginTopic)
+
+    val committerConfigs = List(
+      createCommitterConfig(InitialOffset.latest, "kafkaCommitter", topicsRegexRaw=None, topicsSet=Some(Set("performance"))),
+      createCommitterConfig(InitialOffset.latest, "esCommitter", topicsRegexRaw=None, topicsSet=Some(Set("performance"))),
+      createCommitterConfig(InitialOffset.latest, "falseCommitter", topicsRegexRaw=None, topicsSet=Some(Set("performance"))))
+
+    val taskManager = getTasksManager(committerConfigs)
+
+    "filter out topics as specified by the committer" in {
+      taskManager.matchTopicsRegex("""[A-Za-z]+""".r, performanceTopic, "esCommitter", Set("kafkaCommitter")) shouldEqual
+        Set("kafkaCommitter", "esCommitter")
+      taskManager.matchTopicsRegex("""[A-Za-z]+""".r, performanceTopic, "falseCommitter", Set("kafkaCommitter")) shouldEqual
+        Set("kafkaCommitter")
+    }
+  }
+
+  "matchTopicsSet" should{
+    val performanceTopic = KafkaTopic("performance", Set(0, 1))
+    val loginTopic = KafkaTopic("login", Set(8, 9))
+    val kafkaTopics = Set(performanceTopic, loginTopic)
+
+    val committerConfigs = List(
+      createCommitterConfig(InitialOffset.latest, "kafkaCommitter", topicsRegexRaw=None, topicsSet=Some(Set("performance"))),
+      createCommitterConfig(InitialOffset.latest, "esCommitter", topicsRegexRaw=None, topicsSet=Some(Set("performance"))),
+      createCommitterConfig(InitialOffset.latest, "falseCommitter", topicsRegexRaw=None, topicsSet=Some(Set("performance"))))
+
+    val taskManager = getTasksManager(committerConfigs)
+    "filter out topics as specified by the committer" in {
+      taskManager.matchTopicsSet(Set("performance", "login"),performanceTopic,"esCommitter", Set("kafkaCommitter")) shouldEqual Set("kafkaCommitter", "esCommitter")
+      taskManager.matchTopicsSet(Set("performance", "login"),performanceTopic,"falseCommitter", Set("kafkaCommitter")) shouldEqual Set("kafkaCommitter")
+    }
+  }
+
```

Comment:
can you add one more test
where the there is a "split" acceptance committer
on some topics
where it accepts some but not others
i.e. if the topics to subscribe are Foo, Bar, Baz
the topic regex is “.*”
so all match
but then the committer returns “true” for Foo,
but “false” for Bar, Baz
right now this case does not cover the split it doesn’t cover the split case

the same thing goes for TopicSet